### PR TITLE
chore(ci): Bump actions/labeler from 4 to 5

### PR DESCRIPTION
**Pull request by yin1999**
### Description

Bump actions/labeler from 4 to 5.

The new [input: `sync-label`](https://github.com/actions/labeler?tab=readme-ov-file#inputs) is added to automatically remove unmatched labels (which is configured in label matching entries)

### Motivation

[All the actions which is running on Node 16 will be migrated to Node 20](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/). And the `actions/labeler@v4` is running on Node 16 which may not be supported in the future.

### Related issues and pull requests

downstream PR: mdn/translated-content#18402

testings:

- `system` label: yin1999/content#9
- `Content:Learn` label: yin1999/content#10
- `Content:Firefox` label: yin1999/content#11
- `Content:Other` label: yin1999/content#12

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -1,106 +1,180 @@
...
+          - "!files/en-us/learn/css/**"
+          - "!files/en-us/learn/server-side/django/**"
+          - "!files/en-us/learn/server-side/express_nodejs/**"
+          - "!files/en-us/learn/forms/**"
+          - "!files/en-us/learn/tools_and_testing/github/**"
+          - "!files/en-us/learn/html/**"
+          - "!files/en-us/learn/javascript/**"
+  - changed-files:
+      - any-glob-to-any-file:
+          - files/en-us/web/tutorials/**
```
Comment (yin1999) [inline .github/labeler.yml:74]: If any of the following condition matches, the label will be added:

1. one of the changed files matches `files/en-us/learn/**` and does not match any specified prefixes (`accessibility`, `tools_and_testing/client-side_javascript_frameworks`, etc.)
2. one of the changed files matches `files/en-us/web/tutorials/**`

File: .github/workflows/pr-labeler.yml [hunk 2]
```diff
@@ -16,9 +16,10 @@ jobs:
     name: Label by path
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/labeler@v4
+      - uses: actions/labeler@v5
         with:
           repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          sync-labels: true
```
Comment (yin1999) [inline .github/workflows/pr-labeler.yml:22]: reference: https://github.com/mdn/translated-content/pull/18402#issuecomment-1983470707

**Review by bsmth — APPROVED** [review]
LGTM, it's the same structural change as in https://github.com/mdn/translated-content/pull/18402/files, which is working as expected after landing. Thank you!

**Comment by yin1999** [comment]
Is there anything blocking this :)

**Comment by bsmth** [comment]
> Is there anything blocking this :)

I was waiting for core-dev to +1, but the changes look good (the test PRs are really helpful, ty). I think we can merge shortly!

**Comment by bsmth** [comment]
Thank you @yin1999 for the improvements 🎉

**Pull request merged**